google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -202,16 +254,44 @@ public void submit() {
         span.setStatus(Status.UNKNOWN.withDescription(ex.getMessage()));
         throw translate(ex);
       } finally {
+        if (failedBatches.size() > 0) {
+          processBatchRequest();
+          throw new StorageException(500, "batchError");
+        }
         scope.close();
         span.end(HttpStorageRpcSpans.END_SPAN_OPTIONS);
       }
     }
+
+    private void processBatchRequest() {
+      if (failedBatches.size() > 0) {
```

Comment:
this `if` is redundant, it complicates the code - adds an extra level

Reply:
Done.